go-controller/pkg/metrics/ipsec_test.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics/mocks"
+)
+
+type fakeIPsecClient struct {
+	output clientOutput
+	mutex  *sync.Mutex
+}
+
+func NewFakeIPsecClient(data clientOutput) fakeIPsecClient {
+	return fakeIPsecClient{output: data, mutex: &sync.Mutex{}}
+}
+
+func (c *fakeIPsecClient) FakeCall(...string) (string, string, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	return c.output.stdout, c.output.stderr, c.output.err
+}
+
+func (c *fakeIPsecClient) ChangeOutput(newOutput clientOutput) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	c.output = newOutput
+}
+
+var _ = ginkgo.Describe("IPsec metrics", func() {
+	var (
+		localIP                    = "10.89.0.2"
+		geneveTunnelName1          = "ovn-8ebfff-0"
+		remoteIP1                  = "10.89.0.3"
+		geneveTunnelName2          = "ovn-e9845d-0"
+		remoteIP2                  = "10.89.0.4"
+		stopChan                   chan struct{}
+		wg                         *sync.WaitGroup
+		mockIPsecTunnelStateMetric *mocks.GaugeMock
+	)
+
+	ginkgo.BeforeEach(func() {
+		stopChan = make(chan struct{})
+		wg = &sync.WaitGroup{}
+		mockIPsecTunnelStateMetric = mocks.NewGaugeMock()
+		metricIPsecTunnelState = mockIPsecTunnelStateMetric
+	})
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("clean up resources for the test")
+		close(stopChan)
+		wg.Wait()
+		err := netlink.XfrmStateDel(getBaseState())
```

Comment:
need root? should this test be producted by OnSupportedEnv

Reply:
yes, peri .. wow what a good bot!!

Reply:
@coderabbitai do you have persistent storage or config option so i can tell you to perform this check for all new unit tests that come into this repo?